src/mixer.cr

```diff
@@ -40,17 +40,15 @@ module SDL
 
     # load long sound file
     def self.load_music(filename, type : Type? = nil)
-      #SDL::RWops.open(filename, "rb") do |rwops|
-      #  if type
-      #    audio = LibMixer.load_mus_type_rw(rwops, type.to_s, 1)
-      #    raise SDL::Error.new("Mix_LoadMUSType_RW") unless audio
-      #  else
-      #    audio = LibMixer.load_mus_rw(rwops, 1)
-      #    raise SDL::Error.new("Mix_LoadMUS_RW") unless audio
-      #  end
-      #  audio
-      #end
-      LibMixer.load_mus(filename)
+      rwops = LibSDL.rw_from_file(filename, "rb")
+      if type
+        audio = LibMixer.load_mus_type_rw(rwops, type.to_s, 1)
+        raise SDL::Error.new("Mix_LoadMUSType_RW") unless audio
+      else
+        audio = LibMixer.load_mus_rw(rwops, 1)
+        raise SDL::Error.new("Mix_LoadMUS_RW") unless audio
+      end
+      audio
```

Comment:
Well, the 2 main things are `mus` and `wav`. `mus` is something longer like a song. `wav` is something shorter like a sound. `channels` just refers to like "left" or "right" channel for stereo or mono, etc...

The main thing is deciding if we use something like `SDL::Sample.new(some_filename)` are we making an object to surround just a song, or do have an optional overload like `SDL::Sample.new(some_filename, :wav)` to allow for either one? I'm not too familiar with how the GC works, so from what you can tell, does the mus portion fall in to the same issue of possibly being GC'd? My guess is not in which case we could just wrap `mus` for now.